servers/src/main/java/tachyon/master/file/meta/InodeTree.java

```diff
@@ -266,11 +262,20 @@ public CreatePathResult createPath(TachyonURI path, CreatePathOptions options)
 
     // Create the final path component. First we need to make sure that there isn't already a file
     // here with that name. If there is an existing file that is a directory and we're creating a
-    // directory, nothing needs to be done.
+    // directory, update persistence property of the directories if needed, otherwise, nothing needs
+    // to be done.
     Inode lastInode = currentInodeDirectory.getChild(name);
     if (lastInode != null) {
       if (lastInode.isDirectory() && options.isDirectory()) {
-        return new CreatePathResult();
+        if (!lastInode.isPersisted() && options.isPersisted()) {
+          lastInode.setPersisted(true);
+          traversalResult.getPersisted().add(lastInode);
+          String ufsPath = mMountTable.resolve(getPath(lastInode)).getPath();
+          UnderFileSystem ufs = UnderFileSystem.get(ufsPath, MasterContext.getConf());
+          ufs.mkdirs(ufsPath, false);
+        }
+        return new CreatePathResult(Collections.<Inode>emptyList(), createdInodes,
```

Comment:
The modified Inode will be null if no new Inodes are created, or be the last traversed Inode. You remind me to change the signature from `List<Inode> modifiedInodes` to `Inode modifiedInode`.